### DISCUSSION TITLE
Can I invoke Python3 on WSL from Windows side?

### DISCUSSION THREAD

**Discussion by networks1** [microsoft/WSL#6508] [Q&A]
People on Microsoft Q&A sent me here.  I use Komodo IDE (from Activestate). You configure its debugger by pointing it an interpreter, in this case Python3.

I have a Python library that is not compatible with the Windows version of Python3, but I have it installed on Python3 on WSL. I want to use Komodo to develop/debug scripts using this library. There is a Linux version of Komodo, but there is no X support in WSL (yet).

So I am wondering, is there some way I can use Komodo on the Windows side, but point its debugger at the "remote" interpreter on the WSL side?  Komodo does support [debugging on remote interpreters](https://docs.activestate.com/komodo/12/manual/debugger.html#debugger-proxy).  I suppose I would need to have WSL running to do this, but I don't know how to point to the Python interpreter running on it.

**Comment by markko1** [comment]
Actually it is possible to run Linux GUI programs from WSL2. I do it everyday for my day job. Just google it :)
I personally like X410. It is not free, but it's cheap and their website has perfect tutorials on how to set it up.

**Reply by networks1** [reply]
Dude, thanks for the tip!  Installed X410 and it worked on the first try.  It was also on sale for $9.99

**Reply by Biswa96** [reply]
[VcXsrv][1] is free a software (free in speech and in beer).

[1]: https://sourceforge.net/projects/vcxsrv/

**Comment by Biswa96** [comment]
> Can I invoke Python3 on WSL from Windows side?

```bat
wsl.exe python -c "print('hello')"
```

More info at https://docs.microsoft.com/en-us/windows/wsl/interop

**Reply by networks1** [reply]
Works, but I need a way to connect it to the Komodo debugger.

**Reply by azinsharaf** [reply]
@networks1 Hi Steve, I have similar question but vis versa. Can I invoke Windows python3 interpreter from the WSL side?
I have a python interpreter (a conda env) in Windows that can't be built in Linux. I want to use WSL and either neovim (my preference) or VS Code  and be able to use the Win python interpreter. As far as I know none of the mentioned tools can do it. Hope someone has the answer.
